lib/jekyll-remote-theme/downloader.rb

```diff
@@ -91,8 +126,8 @@ def unzip
       def zip_url
         @zip_url ||= Addressable::URI.new(
           :scheme => theme.scheme,
-          :host   => "codeload.#{theme.host}",
-          :path   => [theme.owner, theme.name, "zip", theme.git_ref].join("/")
+          :host   => theme.host,
+          :path   => [theme.owner, theme.name, "archive", theme.git_ref + ".zip"].join("/")
```

Comment:
According to the GitHub API documentation, host "api.github.com" and rest endpoint "zipball" is the [proper endpoint](https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#download-a-repository-archive-zip) for getting the repository archive zip file.  As per the documentation, this requires support for redirections.  

Using "archive" should extend the the support to other git private repository such as gitlab and still work for github repos.